utils/src/lib.rs

```diff
@@ -17,8 +20,73 @@ use std::{
     env::VarError,
 };
 
+use std::{fmt, path::PathBuf, str::FromStr};
+
 use tokio::runtime::{Handle, Runtime};
 
+#[derive(Debug)]
+pub struct CastAccount {
+    pub address: Address,
+    pub path: Option<DerivationPath>,
+    pub kind: CastAccountType,
+}
+
+#[derive(Debug)]
+pub enum CastAccountType {
+    Keystore,
+    Ledger,
+    Trezor,
+    Remote,
+}
+
+impl fmt::Display for CastAccount {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match &self.path {
+            Some(p) => write!(f, "{:?} {}-{}", self.address, self.kind, p.derivation_string()),
+            _ => write!(f, "{:?} {}", self.address, self.kind),
+        }
+    }
+}
+
+impl fmt::Display for CastAccountType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            CastAccountType::Keystore => write!(f, "keystore"),
+            CastAccountType::Ledger => write!(f, "ledger"),
+            CastAccountType::Trezor => write!(f, "trezor"),
+            CastAccountType::Remote => write!(f, "remote"),
+        }
+    }
+}
+
+pub async fn get_default_keystore() -> Option<PathBuf> {
+    let home = dirs_next::home_dir()?;
+    let mut path = std::path::Path::new(&home).to_path_buf();
+
+    match std::env::consts::OS {
+        "macos" => path = path.join("Library").join("Ethereum"),
+        "windows" => {
+            if let Ok(local_app_data) = std::env::var("LOCALAPPDATA") {
+                path = std::path::Path::new(&local_app_data).join("Ethereum");
+            }
+        }
+        "linux" | _ => path = path.join(".ethereum"),
+    }
+
+    path = path.join("keystore");
+
+    Some(path)
+}
+
```

Comment:
This `get_account` function should be renamed or refactored when trezor support is added